components/dropdown/dropdown-popover-mixin.js

```diff
@@ -0,0 +1,342 @@
+ 			 */
+			trapFocus: { type: Boolean, attribute: 'trap-focus' },
+			/**
+			 * Provide custom offset, positive or negative
+			 * @type {string}
+			 */
+			verticalOffset: { type: String, attribute: 'vertical-offset' },
+			_blockEndScroll: { state: true },
+			_blockStartScroll: { state: true },
+			_dropdownContent: { type: Boolean, attribute: 'dropdown-content', reflect: true },
+			_hasFooterSlotContent: { state: true },
+			_hasHeaderSlotContent: { state: true }
+		};
+	}
+
+	static get styles() {
+		return [super.styles, css`
+			.dropdown-content-layout {
+				align-items: flex-start;
+				display: flex;
+				flex-direction: column;
+			}
+			.dropdown-content {
+				box-sizing: border-box;
+				flex: auto;
+				max-width: 100%;
+				overflow-y: auto;
+				padding: 1rem;
+			}
+			.dropdown-header,
+			.dropdown-footer {
+				box-sizing: border-box;
+				flex: none;
+				max-width: 100%;
+				min-height: 5px;
+				padding: 1rem;
+				position: relative;
+				width: 100%;
+				z-index: 2;
+			}
+			.dropdown-header {
+				border-block-end: 1px solid var(--d2l-popover-border-color, var(--d2l-popover-default-border-color));
+				border-start-end-radius: var(--d2l-popover-border-radius, var(--d2l-popover-default-border-radius));
+				border-start-start-radius: var(--d2l-popover-border-radius, var(--d2l-popover-default-border-radius));
+			}
+			.dropdown-footer {
+				border-block-start: 1px solid var(--d2l-popover-border-color, var(--d2l-popover-default-border-color));
+				border-end-end-radius: var(--d2l-popover-border-radius, var(--d2l-popover-default-border-radius));
+				border-end-start-radius: var(--d2l-popover-border-radius, var(--d2l-popover-default-border-radius));
+			}
+			.dropdown-no-header,
+			.dropdown-no-footer {
+				display: none;
+			}
+			.dropdown-no-padding {
+				padding: 0;
+			}
+			.dropdown-header-scroll {
+				box-shadow: 0 3px 3px 0 var(--d2l-popover-shadow-color, var(--d2l-popover-default-shadow-color));
+			}
+			.dropdown-footer-scroll {
+				box-shadow: 0 -3px 3px 0 var(--d2l-popover-shadow-color, var(--d2l-popover-default-shadow-color));
+			}
+
+			:host([_mobile][_mobile-tray-location="inline-start"][opened]) .dropdown-content-layout,
+			:host([_mobile][_mobile-tray-location="inline-end"][opened]) .dropdown-content-layout {
+				height: 100vh;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.opened = false;
+		this.noAutoClose = false;
+		this.noAutoFit = false;
+		this.noAutoFocus = false;
+		this.noPadding = false;
+		this.noPaddingFooter = false;
+		this.noPaddingHeader = false;
+		this.noPointer = false;
+		this.trapFocus = false;
+
+		this._blockEndScroll = false;
+		this._blockStartScroll = false;
+		this._dropdownContent = true;
+		this._hasFooterSlotContent = false;
+		this._hasHeaderSlotContent = false;
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		this.#contentElement = this.shadowRoot?.querySelector('.dropdown-content');
```

Comment:
This is an optimization same as the previous code. I think it is to avoid querying the DOM hundreds of times while scrolling, resizing, etc.